### PR TITLE
Bugfix FXIOS-14659 [Relay] Fix accessory views in PBM

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-14659)

## :bulb: Description

Fixes issue with accessory views not appearing in PBM.

## :pencil: Checklist
- [x] I filled in the ticket numbers and a description of my work
- [x] I updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] I ensured unit tests pass and wrote tests for new code
- [ ] If working on UI, I checked and implemented accessibility (Dynamic Text and VoiceOver)
- [ ] If adding telemetry, I read the [data stewardship requirements](https://github.com/mozilla-mobile/firefox-ios/wiki/Adding-Glean-Telemetry-Events) and will request a data review
- [ ] If adding or modifying strings, I read the [guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/How-to-add-and-modify-Strings) and will request a string review from l10n
- [ ] If needed, I updated documentation and added comments to complex code

### CODE REVIEW

**Comment by mattreaganmozilla** [comment]
@PARAIPAN9 Thanks so much for investigating this. I'm doing some additional testing but LMK if you have any additional concerns or think this area needs any other updates.

**Comment by PARAIPAN9** [comment]
Tested on both iPad and iPhone and haven't seen any issues.

**Review by PARAIPAN9 — APPROVED** [review]

**Comment by mattreaganmozilla** [comment]
Looks like CI failed on `testVerifyUpdatedPasswordIsSaved()`, going to retry Bitrise as a sanity check.

File: firefox-ios/Client/TabManagement/Tab.swift [hunk 1]
```diff
@@ -1008,13 +1008,8 @@ extension Tab: TabWebViewDelegate {
...
 
     func tabWebViewShouldShowAccessoryView(_ tabWebView: TabWebView) -> Bool {
-        // Hide the default WKWebView accessory view panel for PDF documents and
-        // there is no accessory view to display (but only for iPad cases)
-        let isPDF = mimeType == MIMEType.PDF
-        if UIDevice.current.userInterfaceIdiom == .pad {
-            return !isPDF && tabWebView.accessoryView.hasAccessoryView
-        }
-        return !isPDF
+        // Hide the default WKWebView accessory view panel for PDF documents.
```
Comment (lmarceau) [inline firefox-ios/Client/TabManagement/Tab.swift:1011]: Can we write a unit test for this?

Comment (OrlaM) [reply firefox-ios/Client/TabManagement/Tab.swift:1011]: @mattreaganmozilla you forgot to address this comment. If you don't intend to accept the requested change it is expected that you respond with an explanation or have a discussion about it.

Comment (mattreaganmozilla) [reply firefox-ios/Client/TabManagement/Tab.swift:1011]: Apologies, didn't see this comment yesterday. I don't know of a meaningful unit test for this delegate hook, since the method has now been reduced to:

`return mimeType != MIMEType.PDF`

We could add a unit test to duplicate that logic, but realistically it would just mean future changes would end up being maintained in two different places. LMK if you had something specific in mind for testing, however, or if I'm misunderstanding; I'm happy to open a follow-up PR if needed.

Comment (OrlaM) [reply firefox-ios/Client/TabManagement/Tab.swift:1011]: You can load a PDF into a Tab and then test this function. Then the same test loading anything else in the Tab. PDF is an oft overlooked use case during development. This adds protection should unexpected changes be made to Tab or (TabWebView in particular) in future which is pretty valuable given the 10 mins it should take to write. It's probably more of an integration test if you are using a real TabWebView under the hood, but unit or integration doesn't really matter as long as the use case is tested.

Comment (mattreaganmozilla) [reply firefox-ios/Client/TabManagement/Tab.swift:1011]: I'm happy to take a look, offhand I think we'd want an end-to-end/UI test for that, I don't think a unit or integration test would work. Actually, if you have a few minutes some afternoon for a really quick call LMK, I think it'd be helpful for me to understand the purpose of the test and what you had in mind.
